### DISCUSSION TITLE
Методические рекомендации по CodeSystem

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#249] [Минимальный набор артефактов]
//--------------------------------------
//страница: Методические рекомендации по написанию артефактов на FSH / Системы кодирования
//--------------------------------------

Описание системы кодирования должно содержать описание ресурса CodeSystem (справочника), а также соответствующего ему экземпляра NamingSystem.

Идентификаторы системы кодирования, как основной (предпочтительный), так и дополнительные, описываются в ресурсе NamingSystem.

В описании CodeSystem атрибут ^identifier не указывается.

Верно ли это для случая, если основной идентификатор справочника не совпадает с его каноническим url ???

**Comment by eugenekogan-spb** [comment]
2. При написании систем кодирования для RuCore я указывал атрибут ^content в зависимости от наличия фактических строк справочника в артефакте следующим образом:
- если строк нет, например, [Справочник медицинских организаций](https://fhir-ru.zendoc.me/core.code-systems.nsi-identity-document): not-present
- если включены все варианты, например [Источники оплаты медицинской помощи](https://fhir-ru.zendoc.me/core.code-systems.core-cs-nsi-sources-of-payment) : complete
- если произвольная выборка, например [Документы удостоверяющие личность](https://fhir-ru.zendoc.me/core.code-systems.nsi-identity-document): example

Корректно ли это?

**Reply by eugenekogan-spb** [reply]
@Vrodvi

**Reply by Vrodvi** [reply]
Да, как уже обсуждали вот тут https://github.com/fhir-ru/core/discussions/241#discussioncomment-8663149 неправильно будет использовать только вариант "fragment".

Варианты not-present, complete, example - да они зависят только он включения концептов (строк справочника), вы верно это описали.

При этом у концетов в CodeSystem могут быть не только code и display, которые минимально необходимы для описания концепта, но также могут быть и дополнительные атрибуты (которые отмечаются в атрибуте-массиве [property](https://www.hl7.org/fhir/codesystem-definitions.html#CodeSystem.property)). Это дополнительные атрибуты не влияют на выбор вариантов  not-present, complete, example. Т.е. может быть описан CodeSystem с типом наполнения complete, но для концетов которого вообще не будет указано ни одно дополнительного атрибута. 
Но вот описание (не значения, а факт их наличия, тип данных и предназначение) этих дополнительных атрибутов возможно было бы неплохо включать в описание CodeSystem, т.к. это даст дополнительную информативную ценность - поскольку не всегда понятно как именно дополнительные атрибуты исходного справочника должны быть перенесены в формат FHIR (как именно должны называться эти атрибуты, какие у них должны быть типы данных).

Пример описания дополнительных атрибутов концетов - см. http://hl7.org/fhir/codesystem-fhir-types.html , визуально в руководстве это будет выглядеть так:
![image](https://github.com/fhir-ru/core/assets/92928137/0f1e63b2-05cb-402f-b310-40db72762a61)

Так что часть про описание атрибутов - это на дальнейшее обсуждение.
